compiler/rustc_trait_selection/src/traits/project.rs

```diff
@@ -321,6 +323,32 @@ impl<'a, 'b, 'tcx> TypeFolder<'tcx> for AssocTypeNormalizer<'a, 'b, 'tcx> {
         self.selcx.tcx()
     }
 
+    #[instrument(level = "debug", skip(self))]
+    fn fold_binder<T>(&mut self, t: ty::Binder<'tcx, T>) -> ty::Binder<'tcx, T>
+    where
+        T: TypeFoldable<'tcx>,
+    {
+        let infcx = self.selcx.infcx();
+        let bound_vars = t.bound_vars();
+        let (with_placeholders, mapped_regions, mapped_types, mapped_consts) =
+            infcx.replace_bound_vars_with_placeholders_mapped(t);
+        debug!(?with_placeholders);
+        let folded_with_placeholders = with_placeholders.super_fold_with(self);
```

Comment:
I never implemented the "normalize to an inference variable then quantify the bound vars over the placeholders when the type is fully known" scheme, but seems like it's not needed?